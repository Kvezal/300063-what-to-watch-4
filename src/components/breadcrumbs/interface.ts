export interface IBreadcrumb {
  name: string;
  href?: string;
}

export interface IBreadcrumbsProps {
  breadcrumbs: IBreadcrumb[];
}
