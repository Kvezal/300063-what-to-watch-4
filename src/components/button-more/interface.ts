interface IButtonMoreProps {
  hide: boolean;
  onButtonClick: () => void;
}


export {
  IButtonMoreProps,
};
