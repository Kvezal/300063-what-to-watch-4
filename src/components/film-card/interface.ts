import {ReactNode} from "react";


export interface IFilmCardProps {
  filmId: number;
  filmName: string;
  onCardClick: (filmId: number) => void;
  onPlayingChange: (isPlaying: boolean) => void;
  renderPlayer: () => ReactNode;
}
