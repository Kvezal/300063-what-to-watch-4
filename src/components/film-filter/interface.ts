export interface IFilmFilterProps {
  tabs: string[];
  onItemClick: () => void;
  baseURI: string;
  activeItem: string;
}
