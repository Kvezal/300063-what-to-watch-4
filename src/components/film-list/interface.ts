import {FilmInterface} from "@common/types";


interface IFilmListProps {
  list: FilmInterface[];
  onCardClick: (filmId: number) => void;
  pack?: number;
  step?: number;
}

export {
  IFilmListProps,
};
