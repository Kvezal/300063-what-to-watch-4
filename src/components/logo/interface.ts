export interface ILogoProps {
  linkClass?: {
    [key: string]: boolean;
  };
}
