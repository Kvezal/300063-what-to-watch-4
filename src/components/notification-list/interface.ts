import {INotification} from "@store/notification/interface";


export interface INotificationList {
  notifications: INotification[];
  onCloseNotification: (notificationId: string) => void;
  maxItems: number;
}
