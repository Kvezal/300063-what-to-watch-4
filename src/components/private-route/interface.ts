import * as React from "react";
import {RouteProps} from "react-router-dom";


interface IPrivateRoute {
  path: string;
  exact: boolean;
  render: (props: RouteProps) => React.ReactNode;
  authorizationStatus: string;
}

type TPrivateRoute = RouteProps & IPrivateRoute;

export {
  IPrivateRoute,
  TPrivateRoute,
};
