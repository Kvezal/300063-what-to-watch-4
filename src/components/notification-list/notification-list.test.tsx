import * as React from "react";
import * as render from "react-test-renderer";
import {configure, shallow} from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";

import NotificationList from "./notification-list";
import {EHTTPMethod, ENotificationType} from "@store/notification/interface";


configure({
  adapter: new Adapter(),
});

const notifications = [
  {
    id: `test-id-1`,
    type: ENotificationType.ERROR,
    name: `test`,
    method: EHTTPMethod.GET,
    title: `title 1`,
    text: `title 1`,
  }, {
    id: `test-id-2`,
    type: ENotificationType.ERROR,
    name: `test`,
    method: EHTTPMethod.GET,
    title: `title 2`,
    text: `title 2`,
  }, {
    id: `test-id-3`,
    type: ENotificationType.ERROR,
    name: `test`,
    method: EHTTPMethod.GET,
    title: `title 3`,
    text: `title 3`,
  }, {
    id: `test-id-4`,
    type: ENotificationType.ERROR,
    name: `test`,
    method: EHTTPMethod.GET,
    title: `title 4`,
    text: `title 4`,
  }, {
    id: `test-id-5`,
    type: ENotificationType.ERROR,
    name: `test`,
    method: EHTTPMethod.GET,
    title: `title 5`,
    text: `title 5`,
  }
];

describe(`NotificationListComponent`, () => {
  test(`should render component`, () => {
    const tree = render.create(
        <NotificationList
          list={notifications}
          onCloseNotification={() => null}
          maxItems={5}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`should create component`, () => {
    const notificationListComponent = shallow(
        <NotificationList
          list={notifications}
          onCloseNotification={() => null}
          maxItems={5}
        />
    );
    const notificationList = notificationListComponent.find(`ul.notification-list`);
    expect(notificationList).toHaveLength(1);
  });

  test(`should have notifications`, () => {
    const notificationListComponent = shallow(
        <NotificationList
          list={notifications}
          onCloseNotification={() => null}
          maxItems={5}
        />
    );
    const notificationItems = notificationListComponent.find(`.notification-list__item`);
    expect(notificationItems).toHaveLength(notifications.length);
  });

  test(`should limit notification count`, () => {
    const maxItems = 3;
    const notificationListComponent = shallow(
        <NotificationList
          list={notifications}
          onCloseNotification={() => null}
          maxItems={maxItems}
        />
    );
    const notificationItems = notificationListComponent.find(`.notification-list__item`);
    expect(notificationItems).toHaveLength(maxItems);
  });
});
