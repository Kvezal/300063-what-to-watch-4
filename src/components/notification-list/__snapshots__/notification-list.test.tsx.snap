// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationListComponent should render component 1`] = `
<ul
  className="notification-list"
>
  <li
    className="notification-list__item notification-list__item--error"
  >
    <button
      className="notification-list__button"
      onClick={[Function]}
    >
      Close notification
    </button>
    <h3
      className="notification-list__title"
    >
      title 1
    </h3>
    <p
      className="notification-list__text"
    >
      title 1
    </p>
  </li>
  <li
    className="notification-list__item notification-list__item--error"
  >
    <button
      className="notification-list__button"
      onClick={[Function]}
    >
      Close notification
    </button>
    <h3
      className="notification-list__title"
    >
      title 2
    </h3>
    <p
      className="notification-list__text"
    >
      title 2
    </p>
  </li>
  <li
    className="notification-list__item notification-list__item--error"
  >
    <button
      className="notification-list__button"
      onClick={[Function]}
    >
      Close notification
    </button>
    <h3
      className="notification-list__title"
    >
      title 3
    </h3>
    <p
      className="notification-list__text"
    >
      title 3
    </p>
  </li>
  <li
    className="notification-list__item notification-list__item--error"
  >
    <button
      className="notification-list__button"
      onClick={[Function]}
    >
      Close notification
    </button>
    <h3
      className="notification-list__title"
    >
      title 4
    </h3>
    <p
      className="notification-list__text"
    >
      title 4
    </p>
  </li>
  <li
    className="notification-list__item notification-list__item--error"
  >
    <button
      className="notification-list__button"
      onClick={[Function]}
    >
      Close notification
    </button>
    <h3
      className="notification-list__title"
    >
      title 5
    </h3>
    <p
      className="notification-list__text"
    >
      title 5
    </p>
  </li>
</ul>
`;
